selectors/general.ts

```diff
@@ -18,17 +20,18 @@ export function areTimezonesEnabledAndSupported(state) {
     return config.ExperimentalTimezone === 'true';
 }
 
-export function getBasePath(state) {
+export function getBasePath(state: GlobalState) {
     const config = getConfig(state) || {};
 
     if (config.SiteURL) {
         return new URL(config.SiteURL).pathname;
     }
 
-    return window.basename || '/';
+    return (window as any).basename || '/';
```

Comment:
We can add a type declaration here for the `window` to avoid the any cast, it looks like this:
```
declare global {
    interface Window {
        basename: string
    }
}
```